libs/ui/src/signed_hash_validation_button.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import { useQuery } from '@tanstack/react-query';
+import { SignedHashValidationQrCodeValue } from '@votingworks/types';
+
+import { Button } from './button';
+import { Icons } from './icons';
+import { Modal } from './modal';
+import { QrCode } from './qrcode';
+import { Caption, P } from './typography';
+
+const Content = styled.div`
+  display: flex;
+  align-items: center;
+  gap: 0.5em;
+`;
+
+const QrCodeContainer = styled.div`
+  width: 512px;
+`;
+
+const Details = styled.div`
+  display: flex;
+  flex-direction: column;
+`;
+
+export interface SignedHashValidationApiClient {
+  generateSignedHashValidationQrCodeValue: () => Promise<SignedHashValidationQrCodeValue>;
+}
+
+interface SignedHashValidationModalProps {
+  apiClient: SignedHashValidationApiClient;
+  onClose: () => void;
+}
+
+function SignedHashValidationModal({
```

Comment:
This commit is addressing this comment from a past PR: https://github.com/votingworks/vxsuite/pull/5241#discussion_r1712048110